tests/geometry/WorkingGeometry.spec.ts

```diff
@@ -58,16 +59,97 @@ describe('WorkingGeometry', () => {
+            const faces = [new Face([0, 1, 2], normal), new Face([0, 2, 3], normal)];
             const controlPoints = [vec3.fromValues(0, 0, 0)];
             const square = new WorkingGeometry(vertices, faces, controlPoints);
             const bakedSquare = square.bake();
 
-            // Not testing the colors yet since they don't do anything useful.
-            expect(bakedSquare.indices).toEqual([0, 1, 2, 1, 2, 3]);
+            // Not testing the colors yet since they don't do anything useful
+            expect(bakedSquare.indices).toEqual([0, 1, 2, 0, 2, 3]);
             expect(bakedSquare.vertices).toEqual(vertices);
             expect(bakedSquare.normals).toEqual([normal, normal]);
         });
+        it('can bake a WorkingGeometry that has many merged objects', () => {
+            const rootSquare = TestHelper.square();
+            const childSquare1 = TestHelper.square(vec3.fromValues(1, 0, 0));
+            const childSquare2 = TestHelper.square(vec3.fromValues(-1, 0, 0));
+            const grandChildSquare = TestHelper.square(vec3.fromValues(1, 0, 1));
+            rootSquare.merge(childSquare1);
+            rootSquare.merge(childSquare2);
+            childSquare1.merge(grandChildSquare);
+
+            const bakedObject = rootSquare.bake();
+            expect(bakedObject.indices).toEqual([
+                0,
+                1,
+                2,
+                0,
+                2,
+                3,
+                4,
+                5,
+                6,
+                4,
+                6,
+                7,
+                8,
+                9,
+                10,
+                8,
+                10,
+                11,
+                12,
+                13,
+                14,
+                12,
+                14,
+                15
+            ]);
+            expect(bakedObject.vertices).toEqual([
+                vec3.fromValues(0, 0, 0),
+                vec3.fromValues(0, 1, 0),
+                vec3.fromValues(1, 1, 0),
+                vec3.fromValues(1, 0, 0),
+                vec3.fromValues(1, 0, 0),
+                vec3.fromValues(1, 1, 0),
+                vec3.fromValues(2, 1, 0),
+                vec3.fromValues(2, 0, 0),
+                vec3.fromValues(1, 0, 1),
+                vec3.fromValues(1, 1, 1),
+                vec3.fromValues(2, 1, 1),
+                vec3.fromValues(2, 0, 1),
+                vec3.fromValues(-1, 0, 0),
+                vec3.fromValues(-1, 1, 0),
+                vec3.fromValues(0, 1, 0),
+                vec3.fromValues(0, 0, 0)
+            ]);
+            // Normuals should be an array of 8 (indices/3) [0, 0, 1] vectors
+            const indexStride = 3;
+            const normalCount = bakedObject.indices.length / indexStride;
+            const expectedNormals: vec3[] = [];
+            for (let i = 0; i < normalCount; i += 1) {
+                expectedNormals.push(vec3.fromValues(0, 0, 1));
+            }
```

Comment:
Couldn't find a nice one-liner to just create an array of `n` duplicates of an element. Open to suggestions if anybody knows anything.

Reply:
You can import `range` from lodash, and use:

```typescript
const expectedNormals: vec3[] = range(normalCount).map(() => vec3.fromValues(0, 0, 1));
```